tests/integrationtests/tst_perfparser.cpp

```diff
@@ -172,6 +173,99 @@ private slots:
+        const auto noRead = QStringLiteral("no_r_possible");
+        const auto shell = QStandardPaths::findExecutable(QStringLiteral("bash"));
+        if (!shell.isEmpty()) {
+            QProcess::execute(
+                shell, {QLatin1String("-c"), QStringLiteral("rm -rf %1 && touch %1 && chmod a-r %1").arg(noRead)});
+            parser.initParserArgs(noRead);
+
+            QCOMPARE(parsingFailedSpy.count(), 1);
+            message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+            QVERIFY(message.contains(noRead));
+            QVERIFY(message.contains(QLatin1String("not readable")));
+        }
+    }
+
+    void testFileContent()
+    {
+        Settings::instance()->setAppPath(
+            QFileInfo(QStandardPaths::findExecutable(QStringLiteral("true"))).dir().path());
+        Settings::instance()->setExtraLibPaths(QFINDTESTDATA("file_content"));
+        PerfParser parser(this);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QSignalSpy parsingFinishedSpy(&parser, &PerfParser::parsingFinished);
+        QString message;
+
+        // check pre-exported perfparser
+        const auto perfData = QFINDTESTDATA("file_content/true.perfparser");
+        QVERIFY(!perfData.isEmpty() && QFile::exists(perfData));
+        parser.startParseFile(perfData);
+        QVERIFY(parsingFinishedSpy.wait(2000));
+        QCOMPARE(parsingFailedSpy.count(), 0);
+        QCOMPARE(parsingFinishedSpy.count(), 1);
+
+        // check for invalid data -> expected error
+        const auto noPerf = QFINDTESTDATA("tst_perfparser.cpp");
+        QVERIFY(!noPerf.isEmpty() && QFile::exists(noPerf));
+        parser.startParseFile(noPerf);
+        QVERIFY(parsingFailedSpy.wait(1000));
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(QLatin1String("invalid perf data file")));
+
+        // check for PERFv1 -> expected error
+        const auto perf1 = QFINDTESTDATA("file_content/perf.data.true.v1");
+        QVERIFY(!perf1.isEmpty() && QFile::exists(perf1));
+        parser.startParseFile(perf1);
+        QVERIFY(parsingFailedSpy.wait(1000));
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(QLatin1String("invalid perf data file")));
+
+        // check for PERFv2
+        const auto perf2 = QFINDTESTDATA("file_content/perf.data.true.v2");
+        QVERIFY(!perf2.isEmpty() && QFile::exists(perf2));
+        parser.startParseFile(perf2);
+        QVERIFY(parsingFinishedSpy.wait(
+            8000)); // FIXME: this waittime and the one below must be so high as unavailable elf files are checked
+                    // again and again via PerfSymbolTable::registerElf, as seen in the output
+        QCOMPARE(parsingFailedSpy.count(), 0);
+        QCOMPARE(parsingFinishedSpy.count(), 2);
```

Comment:
This is _by design_ a valid perf file which cannot be completely parsed.

Running this results in multiple identical outputs of

> PerfUnwind::MissingElfFile: Could not find ELF file for 64/ld-2.17.so. This can break stack unwinding and lead to missing symbols.

and the test hints at that this is related to this test taking so long (it runs faster if I add that file).

We seem to cache elf files (return of ` PerfSymbolTable::registerElf()`), but only those that we can resolve.
In this case it would be good to also cache "invalid" / empty returns; allowing to also decrease the wait time for those tests here.
l'm copying this piece to #551, as working on that is totally unrelated to this PR.